src/Controller/CheckinRequestController.php

```diff
@@ -128,9 +175,11 @@ protected function initiateCheckinRequest(CheckinRequest $checkinRequest)
         $checkinRequest->create();
 
         // Initiate a job for non-cancellation requests.
-        if (is_null($checkinRequest->getJobId()) && $this->isUseJobService()) {
+        if (is_null($checkinRequest->getCancelRequestId()) && $this->isUseJobService()) {
+            // Create a jobId and assign it to special checkinJobId property
+            // (to signal that job should not be completed later in execution)
             $checkinRequest->setCheckinJobId(JobService::generateJobId($this->isUseJobService()));
-            // Set jobId for proper responses for non-cancellation requests.
+            // Copy jobId to jobId property (which is saved in db):
```

Comment:
This differs from `qa`. In `qa`, we have:
```
$checkinRequest->setCheckinJobId($checkinRequest->getJobId());
```
This is to support another service that handles check in requests being able to set the job id. I think we probably want to keep the version in `qa`, but add a line to generate the job id if it doesn't exist, something like:
```
if (is_null($checkinRequest ->getJobId()) {
  $checkinRequest->setJobId(JobService::generateJobId($this->isUseJobService()));
}
$checkinRequest->setCheckinJobId($checkinRequest->getJobId());
```